lib/Migration/Version000700Date20230916000000.php

```diff
@@ -38,6 +38,10 @@ class Version000700Date20230916000000 extends SimpleMigrationStep {
 			'name' => 'selection',
 			'db_type' => Types::TEXT,
 		],
+		[
+			'name' => 'usergroup',
+			'db_type' => Types::TEXT,
```

Comment:
I think for building more optimal queries it would actually be nice to have a different table format that not stores the entires in json as text but instead something like this

oc_tables_row_cells_usergroups
| id | column_id | row_id | value | value_type |
|---|---|---|---|---|
| 1 | 2 | 9 | alice | 0 |
| 2 | 2 | 9 | bob | 0 |
| 3 | 2 | 10 | engineering | 1 |

This might be more complex to build but we can do performant filtering on large tables by having an index on the value type and its exact match. For the value type using an integer (0 for user, 1 for group) is common from the server sharing API. Probably something to explore further then. Would be curious if @blizzz may have further thoughts on that.

Reply:
So to have a generic solutions that allows value_type in row_cells_xyz tables:

1. A method like  `lib/Migration/Version000700Date20230916000000::createRowValueTable()` or its optional extension to also add a `value_type` column. The `$columns` array could be extended in that class. For new data types of course we need a repair step.
2. Evaluate an abstract subclass of `RowCellSupper` providing support for setting value type (e.g. adding `setValueTypeWrapper()`) or maybe it is sufficient (though not cleaner) to extend the current class, taking the existence of a value_type from the concrete implementation into account.
3.In the same way evaluate extending `RowCellMapperSuper` with `parseValueTypeOutgoing()` and `parseValueTypeIncoming()` although this is unlikely to contain "weird" data. We shall store there constants only (ideally only ints) and it should not be set by direct user input.
4. `Row2Mapper` needs to become aware of possible `value_type` fields, for example in `insert()`, `getFilterExpression()`  (there we already have type-dependent handling) and generally where at least `formatValue()` is being used
5. Insert ability to merge rows in `oc_tables_row_cells_xyz` that share the same column_id and row_id to be able to support multi-values by row. The difficulty lies in having different cell ids per actually set value, so this makes editing a cell difficult, but on a first glance the frontend does not work with cell IDs when updating a value. `RowCellMapperSuper::findByRowAndColumn()` returns a single entity – a starting point would be to have a generic extension for multiple values with a method lile `findMultipleByRowAndColumn()` that can return multiple entities which will later on be merged for the frontend.
6. Where applicable frontend also has to supply value_type information, e.g. when submitting such cell values.

+ Not looked into it yet by me: frontend live filtering, might not be an issue.

A non-generic solutions also ticks almost all items, albeit on concrete implemenations. The value_type *might* come in handy with other future types (could imagine it for columns like file or image picker, ip address, icon/emoji picker).

On a second  look, i just saw the `TextLinkBusiness` thing and it stores already up to three data points in its value (`title`, `value`, `providerId`). If we want to be able to serve scenarios like these, we'd need to look into a more flexible approach.

Reply:
There would be no operation against a single row, that's true. I think the update is already an update-or-insert, so new values should not be problematic, but those that may be delete do require additional checks and delete queries, yes.

Reply:
@blizzz @juliushaertl I updated the table format